### PR TITLE
feat: fetch sub-conversation and group info FS-1805

### PR DESCRIPTION
**Pull request by David-Henner**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1805" title="FS-1805" target="_blank">TaskFS-1805</a>  [iOS] Support fetching subgroup and subgroup info
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This PR adds support for fetching sub-conversations and their group info

### Changes

- Created action and action handler to fetch a given sub-conversation
- Refactored the action and action handler for fetching conversation group info
- Created action and action handler to fetch a sub-conversation's group info

#### Test Coverage (Optional)

- [x] Added tests for action handlers

### Notes

I will change the base branch when the feature branch for sub-conversations is created

### CODE REVIEW

File: wire-ios-data-model/Source/MLS/MLSController.swift [hunk 1]
```diff
@@ -1022,33 +1021,9 @@ public final class MLSController: MLSControllerProtocol {
...
-                domain: domain,
-                context: context
-            )
-
-        } catch let error {
-            self.logger.warn("failed to fetch public group state with error: \(String(describing: error))")
-            throw MLSGroupStateError.failedToFetchGroupState
-        }
-    }
-
```
Comment (David-Henner) [inline wire-ios-data-model/Source/MLS/MLSController.swift]: this was unused

File: wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift [hunk 2]
```diff
@@ -40,15 +55,34 @@ class FetchPublicGroupStateActionHandler: ActionHandler<FetchPublicGroupStateAct
             return nil
```
Comment (johnxnguyen) [inline wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift]: Shouldn't we also fail the action here with `.emptyParameters`?

File: wire-ios/Wire-iOS Tests/ReferenceImages/IncomingConnectionViewTests/testThatItRendersWithUserName.1.png
Comment (johnxnguyen) [inline wire-ios/Wire-iOS Tests/ReferenceImages/IncomingConnectionViewTests/testThatItRendersWithUserName.1.png:1]: Which of these is the correct one?

Comment (David-Henner) [reply wire-ios/Wire-iOS Tests/ReferenceImages/IncomingConnectionViewTests/testThatItRendersWithUserName.1.png:1]: the right one

File: wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift (hunk 2 above)
Comment (David-Henner) [reply wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift]: it's already the case

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios-data-model/Source/MLS/Actions/FetchMLSGroupInfoAction.swift [hunk 3]
```diff
@@ -17,9 +16,42 @@
 
 import Foundation
 
-public final class FetchPublicGroupStateAction: EntityAction {
+public final class FetchMLSConversationGroupInfoAction: BaseFetchMLSGroupInfoAction {
```
Comment (netbe) [inline wire-ios-data-model/Source/MLS/Actions/FetchMLSGroupInfoAction.swift]: Could we move each class to its own file ?

File: wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift [hunk 4]
```diff
@@ -20,18 +20,33 @@ import Foundation
 import WireTransport
 import WireDataModel
 
-class FetchPublicGroupStateActionHandler: ActionHandler<FetchPublicGroupStateAction> {
+class FetchMLSConversationGroupInfoActionHandler: BaseFetchMLSGroupInfoActionHandler<FetchMLSConversationGroupInfoAction> {
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift]: final

File: wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift [hunk 5]
```diff
@@ -20,18 +20,33 @@ import Foundation
...
+        return self.request(
+            for: action,
+            path: "/conversations/\(action.domain)/\(action.conversationId.transportString())/groupinfo",
+            apiVersion: apiVersion,
+            minRequiredAPIVersion: .v3
+        )
+    }
+}
+
+class FetchMLSSubconversationGroupInfoActionHandler: BaseFetchMLSGroupInfoActionHandler<FetchMLSSubconversationGroupInfoAction> {
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift]: move to own class

File: wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift [hunk 6]
```diff
@@ -40,15 +55,34 @@ class FetchPublicGroupStateActionHandler: ActionHandler<FetchPublicGroupStateAct
...
+        return request(
+            for: action,
+            path: "/conversations/\(action.domain)/\(action.conversationId.transportString())/subconversations/\(action.subgroupType)/groupinfo",
+            apiVersion: apiVersion,
+            minRequiredAPIVersion: .v4
+        )
+    }
+}
+
+class BaseFetchMLSGroupInfoActionHandler<T: BaseFetchMLSGroupInfoAction>: ActionHandler<T> {
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/FetchMLSGroupInfoActionHandler.swift]: move to own class

**Review by netbe — APPROVED** [review]
just minor suggestions about moving classes to their own file

**Pull request merged**